### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by iropal**

### CODE REVIEW

File: src/components/components/join-cta/join-cta.js [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
-  const baseClass = 'join-cta';
+  const baseClass = "join-cta";
   // handle component classes. Add the variations found in props.
   let classes = [baseClass];
   if (props.tall) classes.push(`${baseClass}--tall`);
-  if (props.verticalGutters) if (props.className) classes.push(props.className);
+  if (props.className) classes.push(props.className);
 
-  const classString = classes.join(' ');
+  const classString = classes.join(" ");
```
Comment (alexdor) [inline src/components/components/join-cta/join-cta.js]:
```suggestion
  const classString = `${baseClass} ${props.tall ? baseClass + "--tall" : ""} ${props.className}`;
```

File: src/components/components/leader-tools/leader-tools.js [hunk 2]
```diff
@@ -79,12 +94,12 @@ export class LeaderToolList extends React.PureComponent {
...
 
-    const listClasses = ['leader-tools-list__wrapper'];
-    if (snapshot.isOpen) listClasses.push('leader-tools-list__wrapper--open');
+    const listClasses = ["leader-tools-list__wrapper"];
+    if (snapshot.isOpen) listClasses.push("leader-tools-list__wrapper--open");
 
     const hideDetails = !snapshot.isOpen && this.props.isMobile;
     return (
-      <div className={listClasses.join(' ')}>
+      <div className={listClasses.join(" ")}>
```
Comment (alexdor) [inline src/components/components/leader-tools/leader-tools.js]: Maybe using a template literal here would be cleaner as well

File: src/components/components/member-teaser/member-teaser.js [hunk 3]
```diff
@@ -62,9 +61,8 @@ const MemberTeaser = props => {
   const classes = [inheritedClass, "member-teaser"];
   if (isLeader) classes.push("member-teaser--leader");
   return (
-    <ShadowBox smallPaddings className={classes.join(" ")}>
-      <div> {preview} </div>
-      <Link to={`/members/${props.member.id}`} className="member-teaser__inner">
+    <Link to={`/members/${props.member.id}`} className={classes.join(" ")}>
```
Comment (alexdor) [inline src/components/components/member-teaser/member-teaser.js]: A template literal would probably be cleaner here as well

File: src/components/components/organization-teaser/organization-teaser.js [hunk 4]
```diff
@@ -16,9 +18,9 @@ const OrganizationTeaser = props => {
   const url = props.organization.link;
   const icon = props.organization.attachment.publicURL;
 
-  let classes = ['organization-teaser'];
+  let classes = ["organization-teaser"];
   if (props.className) classes.push(props.className);
-  let classString = classes.join(' ');
+  let classString = classes.join(" ");
```
Comment (alexdor) [inline src/components/components/organization-teaser/organization-teaser.js]: A template literal would probably be cleaner here as well

File: src/components/components/organization/organization.js [hunk 5]
```diff
@@ -20,29 +20,32 @@ const Organization = props => {
   const icon = props.organization.attachment.publicURL;
   const description = props.organization.description;
 
-  let classes = ['organization'];
+  let classes = ["organization"];
   if (props.className) classes.push(props.className);
-  let classString = classes.join(' ');
+  let classString = classes.join(" ");
```
Comment (alexdor) [inline src/components/components/organization/organization.js]: A template literal would probably be cleaner here as well

File: src/components/components/toggle/toggle.js [hunk 6]
```diff
@@ -1,14 +1,23 @@
...
+ */
 
 export default props => {
   const onClick = props.onClick;
-  const classes = [props.className, 'toggle'];
-  if (props.active) classes.push('toggle--active');
-  const classString = classes.join(' ');
+  const classes = [props.className, "toggle"];
+  if (props.active) classes.push("toggle--active");
+  const classString = classes.join(" ");
```
Comment (alexdor) [inline src/components/components/toggle/toggle.js]: A template literal would probably be cleaner here as well

File: src/components/components/user-menu/user-menu.js [hunk 7]
```diff
@@ -52,36 +64,44 @@ class UserMenu extends React.PureComponent {
...
     let userId =
       this.props.user.user && this.props.user.user.id
         ? this.props.user.user.id
-        : '';
-    let classes = ['user-menu'];
+        : "";
+    let classes = ["user-menu"];
     if (stateClass) classes.push(stateClass);
-    let classString = classes.join(' ');
+    let classString = classes.join(" ");
```
Comment (alexdor) [inline src/components/components/user-menu/user-menu.js]: A template literal would probably be cleaner here as well

File: src/components/components/user-menu/user-menu.js [hunk 8]
```diff
@@ -90,7 +110,7 @@ class UserMenu extends React.PureComponent {
         onKeyPress={this.handleKeyPress}
         role="button"
         tabIndex={0}
-        aria-controls={'user-menu-wrapper'}
+        aria-controls={"user-menu-wrapper"}
```
Comment (alexdor) [inline src/components/components/user-menu/user-menu.js]: Why is this evaluated? Isn't it just a string?

File: src/components/layouts/layout-5col/layout-5col.js [hunk 9]
```diff
@@ -1,37 +1,49 @@
...
 
-  // add the `Layout-2col-unequal__col` class to every child.
+  /**
+   * Add the `layout-5col__col` class to every child.
+   * Required for the layout to apply correctly.
+   **/
   const children = React.Children.map(props.children, child => {
     const className = `${
-      child.props.className ? child.props.className : ''
+      child.props.className ? child.props.className : ""
```
Comment (alexdor) [inline src/components/layouts/layout-5col/layout-5col.js]:
```suggestion
      child.props.className || ""
```

File: src/components/navigation/footer-navigation/footer-navigation.js [hunk 10]
```diff
@@ -117,9 +151,9 @@ export default class Navigation extends React.PureComponent {
     const snapshot = { ...this.state };
     const links = this.parseLinks(this.props.links);
     const isExpanded = snapshot.isOpen && snapshot.isMobile;
-    let classes = ['footer__section footer__section--mobile'];
+    let classes = ["footer__section footer__section--mobile"];
     if (this.props.className) classes.push(this.props.className);
-    let classString = classes.join(' ');
+    let classString = classes.join(" ");
```
Comment (alexdor) [inline src/components/navigation/footer-navigation/footer-navigation.js]: A template literal would probably be cleaner here as well

File: src/components/components/club-full/club-full.js [hunk 11]
```diff
@@ -41,18 +41,19 @@ function ClubFull(props) {
   useEffect(() => {
     const path = window.location.pathname.split("/");
     const id = (props.club || {}).id || path[path.indexOf("clubs") + 1];
-    getClub(id).then(updateClub);
-  }, []);
+    !editing && getClub(id).then(updateClub);
```
[the remark this answers is not in the document]
Comment (iropal) [reply src/components/components/club-full/club-full.js]: @alexdor without this check the club page is not editable. It creates a bug where it keeps overwriting the user input. This is why I keep adding it.

[the remark this answers is not in the document]
Comment (alexdor) [reply src/components/components/club-full/club-full.js]: The change underneath should solve the bug that you are mentioning

File: src/components/components/club-full/club-full.js [hunk 12]
```diff
@@ -42,17 +42,18 @@ function ClubFull(props) {
     const path = window.location.pathname.split("/");
     const id = (props.club || {}).id || path[path.indexOf("clubs") + 1];
     getClub(id).then(updateClub);
-  }, []);
+  }, [getClub, props.club, updateClub]);
```
Comment (alexdor) [inline src/components/components/club-full/club-full.js]:
```suggestion
  }, []);
```

File: src/components/components/club-full/club-full.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (iropal) [reply src/components/components/club-full/club-full.js]: I does :)

File: src/components/components/club-full/club-full.js (hunk 12 above)
Comment (iropal) [reply src/components/components/club-full/club-full.js]: I found out what's going on with this line.. It is being auto corrected on commit. That's why this code keeps coming back...

Comment (alexdor) [reply src/components/components/club-full/club-full.js]: Add: 
// eslint-disable-next-line react-hooks/exhaustive-deps
before this line

File: src/components/components/header-navigation/header-navigation.js [hunk 13]
```diff
@@ -53,28 +70,42 @@ class HeaderNavigation extends React.PureComponent {
...
+  handleKeyPress = event => {
+    if (event.key === "Enter") return this.handleOpen;
   };
 
   render() {
     const snapshot = { ...this.state };
     const isExpanded = snapshot.open;
-    const stateClass = snapshot.open ? 'is-expanded' : 'is-collapsed';
+    const stateClass = snapshot.open ? "is-expanded" : "is-collapsed";
     const isHidden = !snapshot.open;
```
[the remark this answers is not in the document]
Comment (iropal) [reply src/components/components/header-navigation/header-navigation.js]: In fact it is the same. I simplified the code to make it clear. I'd like to keep the state class for cleaner css.

File: src/components/components/faq-item/faq-item.js [hunk 14]
```diff
@@ -1,36 +1,40 @@
...
-    const stateClass = snapshot.open ? 'is-expanded' : 'is-collapsed';
+    const stateClass = snapshot.open ? "is-expanded" : "is-collapsed";
     const isHidden = !snapshot.open;
     const id = this.props.id;
 
-    let classes = ['faq__item'];
+    let classes = ["faq__item"];
     if (stateClass) classes.push(stateClass);
-    let classString = classes.join(' ');
+    let classString = classes.join(" ");
```
Comment (alexdor) [inline src/components/components/faq-item/faq-item.js]: A template literal would clean this as well

File: src/components/components/faq-item/faq-item.js [hunk 15]
```diff
@@ -1,36 +1,40 @@
...
+  /**
+   * Handles keys.
+   */
   handleKeyPress = e => {
-    if (e.key === 'Enter') return this.handleToggle;
+    if (e.key === "Enter") return this.handleToggle;
   };
 
   render() {
     const snapshot = { ...this.state };
```
Comment (alexdor) [inline src/components/components/faq-item/faq-item.js]:
```suggestion
    const snapshot = this.state;
```

File: src/components/components/header-navigation/header-navigation.js [hunk 16]
```diff
@@ -95,8 +123,9 @@ class HeaderNavigation extends React.PureComponent {
               onClick={this.handleOpen}
               onKeyPress={this.handleKeyPress}
               className="button button--header header__button"
-              aria-controls={'header-navigation-container'}
+              aria-controls={"header-navigation-container"}
```
Comment (alexdor) [inline src/components/components/header-navigation/header-navigation.js]: No need to evaluate this

File: src/components/components/faq-item/faq-item.js [hunk 17]
```diff
@@ -1,36 +1,40 @@
...
     open: false
   };
 
-  // Handles the collapse actions.
+  /**
+   * Handles the collapse/expand actions.
+   */
   handleToggle = () => {
     const snapshot = { ...this.state };
     this.setState({ open: !snapshot.open });
```
Comment (alexdor) [inline src/components/components/faq-item/faq-item.js]:
```suggestion
    this.setState(state => ({ open: !state.open }));
```

File: src/components/components/faq-item/faq-item.js [hunk 18]
```diff
@@ -1,36 +1,40 @@
...
   state = {
     open: false
   };
 
-  // Handles the collapse actions.
+  /**
+   * Handles the collapse/expand actions.
+   */
   handleToggle = () => {
     const snapshot = { ...this.state };
```
Comment (alexdor) [inline src/components/components/faq-item/faq-item.js]:
```suggestion
```

File: src/components/components/leader-tools/leader-tools.js [hunk 19]
```diff
@@ -67,6 +79,9 @@ export class LeaderToolList extends React.PureComponent {
     return button;
   }
 
+  /**
+   * Updates the object open/closed state.
+   */
   handleOpen() {
     const snapshot = { ...this.state };
     this.setState({ isOpen: !snapshot.isOpen });
```
Comment (alexdor) [inline src/components/components/leader-tools/leader-tools.js]:
```suggestion
    this.setState(state => ({ isOpen: !state.isOpen }));
```

File: src/components/components/user-menu/user-menu.js [hunk 20]
```diff
@@ -52,36 +65,40 @@ class UserMenu extends React.PureComponent {
...
+   * Checks if the user is logged in.
+   * @returns {boolean}
+   */
   isUserLoggedIn = () => {
     return !!this.props.user.user;
   };
 
   render() {
-    // copy of the state
     const snapshot = { ...this.state };
```
Comment (alexdor) [inline src/components/components/user-menu/user-menu.js]:
```suggestion
    const snapshot = this.state;
```

File: src/components/navigation/footer-navigation/footer-navigation.js [hunk 21]
```diff
@@ -75,18 +96,31 @@ export default class Navigation extends React.PureComponent {
...
   }
 
-  handleClick = event => {
+  /**
+   * Toggles open / close and updates navigation state.
+   */
+  handleClick = () => {
     const snapshot = { ...this.state };
     if (!snapshot.isMobile) return;
     this.setState({
```
Comment (alexdor) [inline src/components/navigation/footer-navigation/footer-navigation.js]:
```suggestion
    this.setState(({isOpen}) => ({ isOpen: !isOpen})
```

File: src/components/navigation/footer-navigation/footer-navigation.js [hunk 22]
```diff
@@ -75,18 +96,31 @@ export default class Navigation extends React.PureComponent {
     };
   }
 
-  handleClick = event => {
+  /**
+   * Toggles open / close and updates navigation state.
+   */
+  handleClick = () => {
     const snapshot = { ...this.state };
```
Comment (alexdor) [inline src/components/navigation/footer-navigation/footer-navigation.js]:
```suggestion
```

File: src/components/navigation/footer-navigation/footer-navigation.js [hunk 23]
```diff
@@ -75,18 +96,31 @@ export default class Navigation extends React.PureComponent {
...
     if (!snapshot.isMobile) return;
```
Comment (alexdor) [inline src/components/navigation/footer-navigation/footer-navigation.js]:
```suggestion
    if (!this.state.isMobile) return;
```

**Pull request merged**